zhakypadil/Habarla
# WIP Code restructure&address feature

## Ticket
* N/A
## Description
* 1.0.3 Build release.
* Feature added: 
  * Restaurant address link `(only support 2GIS links rn)`
* Bugs Fixed: 
  * User deletion bugs (user wasn't fully deleted due to race conditions and Firebase's `auth/requires-recent-login` error)
  * Bunch of bugs with Customer/Vendor roles, where restaurantId was required for both users
* Major code cleanup, but not perfect yet. `async/await-then and try-catch` are used interchangeably. Better to have some fixed approach
* App icon replaced  

| Old  | New |
| ------------- | ------------- |
| ![icon](https://user-images.githubusercontent.com/32118989/235767472-d802f10a-f5e0-4f3c-90b0-5323cbfac463.png) | ![124048574_padded_logo](https://user-images.githubusercontent.com/32118989/235767646-4521c387-232d-408d-845e-a7f2834124ac.png) |



## Test Plan
### Test 1: Functionality :white_check_mark:
* Add the address to the restaurant
  * Verify it is saved and visible on all following discounts (updating all discounts will be done later)
  * Clicking on Discounts has a response (user interview issue, where everyone wanted clicking to do something)
* User deletion works perfectly
